v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/build-system.ts

```diff
@@ -118,11 +131,44 @@ export class SolidityBuildSystemImplementation implements SolidityBuildSystem {
 
     const compilationJobs = [...new Set(compilationJobsPerFile.values())];
 
-    // TODO: Filter the compilation jobs based on the cache
+    await Promise.all(
+      compilationJobs.map(async (compilationJob) =>
+        compilationJob.getBuildId(),
+      ),
+    );
```

Comment:
```suggestion
    // We precompute the build ids here, which are cached internally in each compilation job
    await Promise.all(
      compilationJobs.map(async (compilationJob) =>
        compilationJob.getBuildId(),
      ),
    );
```

What about adding a comment like this?

Reply:
Of course, done 📖